### PR TITLE
Devengage 1573 implementing tables

### PR DESCRIPTION
**Pull request by Zino-ctrlZ**

### CODE REVIEW

File: app/src/components/formdemo/FormDemo.tsx [hunk 1]
```diff
@@ -750,6 +752,148 @@ export default function FormDemo() {
 		),
 	});
 
+	const tableProps = {
+		className: 'table',
+		headerRow: {
+			cells: [
+				{
+					renderedContent: <p>Observation Type</p>,
```
Comment (timsmithgenesys) [inline app/src/components/formdemo/FormDemo.tsx]: I made a change so it's no longer necessary to do this. You can only provide `content` if you don't want it rendered differently than the simple value. But `renderedContent` will be preferred if it's set.

File: app/src/components/formdemo/FormDemo.tsx [hunk 2]
```diff
@@ -750,6 +752,148 @@ export default function FormDemo() {
...
+					renderedContent: <p>Example Use Case(s)</p>,
+					content: 'Example Use Case(s)',
+					align: 'left',
+				},
+			],
+		},
+		indentation: 0,
+		rows: [
+			{
+				raw: '|[Flow](flows/)|Returns a single metric showing currently observed flow interactions. |**Note**: There is no additional documentation for the flow observations.  The link will take you to the endpoints page for analytics. |\n',
```
Comment (timsmithgenesys) [inline app/src/components/formdemo/FormDemo.tsx]: `DataTableRow` does not have a `raw` property

File: package.json [hunk 3]
```diff
@@ -49,5 +49,9 @@
 		"react-dom": ">=16",
 		"react-syntax-highlighter": "^15.5.0",
 		"uuid": "^9.0.0"
+	},
+	"dependencies": {
+		"moment": "^2.29.4",
```
Comment (timsmithgenesys) [inline package.json:51]: moment should be a peer dependency. postcss is already listed there. As a general rule, packages should never have direct dependencies, only dev and peer dependencies. Direct dependencies will cause rollup to try and package them into the package; we do not want that.

File: src/datatable/DataTable.scss [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+		gap: 8px;
+		width: 100%;
+		box-sizing: border-box;
+
+		.filter-toggle {
+			color: var($--theme-core-link-color);
+			font-size: 12px;
+			line-height: 0;
+			margin: 6px 0 0 10px;
+			// padding: 2px;
```
Comment (timsmithgenesys) [inline src/datatable/DataTable.scss]: remove commented out code

**Review by timsmithgenesys — CHANGES_REQUESTED** [review]

File: app/src/components/formdemo/FormDemo.tsx (hunk 2 above)
Comment (Zino-ctrlZ) [reply app/src/components/formdemo/FormDemo.tsx:779]: DataTableCell has a raw property and DataTableRow is an array of DataTableCell

File: app/src/components/formdemo/FormDemo.tsx (hunk 1 above)
Comment (Zino-ctrlZ) [reply app/src/components/formdemo/FormDemo.tsx]: So content isn't a required prop?

Comment (timsmithgenesys) [reply app/src/components/formdemo/FormDemo.tsx]: The TS definition is correct;`content` is required, `renderedContent` is not. https://github.com/purecloudlabs/genesys-react-components/pull/18/commits/84b1e2418e46fc23cd58c441de33051bf7c4b78c

Comment (Zino-ctrlZ) [reply app/src/components/formdemo/FormDemo.tsx]: Do you prefer it removed in the demo?

Comment (timsmithgenesys) [reply app/src/components/formdemo/FormDemo.tsx]: Yes, demonstrate it the way it's meant to be used. Only provide `content` if it's a simple value. Use `content` and `renderedContent` to demonstrate the case where the content used for sorting/filtering is different from what's displayed.

**Review by timsmithgenesys — APPROVED** [review]

**Pull request merged**